File: library/src/test/java/com/mindera/skeletoid/generic/StringUtilsUnitTests.java
package com.mindera.skeletoid.generic;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class StringUtilsUnitTests {

    @Test
    public void testsOrdinalIndexNullString() {
        assertEquals(-1, StringUtils.ordinalIndexOf(null, "abc", 1));
    }

    @Test
    public void testsOrdinalIndexOfNullSubstring() {
        assertEquals(-1, StringUtils.ordinalIndexOf("abc", null, 1));
    }

    @Test
    public void testsOrdinalIndexOfNullStrings() {
        assertEquals(-1, StringUtils.ordinalIndexOf(null, null, 1));
    }

    @Test
    public void testsOrdinalIndexOfEmptyStrings() {
        assertEquals(-1, StringUtils.ordinalIndexOf("", "", 5));
    }

    @Test
    public void testsOrdinalIndexOfInvalidIndex() {
        assertEquals(-1, StringUtils.ordinalIndexOf("abcabc", "c", -2));
    }

    @Test
    public void testsOrdinalIndexOf() {
        assertEquals(5, StringUtils.ordinalIndexOf("abcabc", "c", 2));
    }
}

File: base/src/test/java/com/mindera/skeletoid/generic/StringUtilsUnitTests.kt
package com.mindera.skeletoid.generic

import com.mindera.skeletoid.generic.StringUtils.ordinalIndexOf
import junit.framework.Assert
import org.junit.Test

class StringUtilsUnitTests {

    @Test
    fun testsOrdinalIndexNullString() {
        Assert.assertEquals(
            -1,
            ordinalIndexOf(null, "abc", 1)
        )
    }

    @Test
    fun testsOrdinalIndexOfNullSubstring() {
        Assert.assertEquals(
            -1,
            ordinalIndexOf("abc", null, 1)
        )
    }

    @Test
    fun testsOrdinalIndexOfNullStrings() {
        Assert.assertEquals(
            -1,
            ordinalIndexOf(null, null, 1)
        )
    }

    @Test
    fun testsOrdinalIndexOfEmptyStrings() {
        Assert.assertEquals(
            -1,
            ordinalIndexOf("", "", 5)
        )
    }

    @Test
    fun testsOrdinalIndexOfInvalidIndex() {
        Assert.assertEquals(
            -1,
            ordinalIndexOf("abcabc", "c", -2)
        )
    }

    @Test
    fun testsOrdinalIndexOf() {
        Assert.assertEquals(
            5,
            ordinalIndexOf("abcabc", "c", 2)
        )
    }
}